napari/layers/utils/text.py

```diff
@@ -0,0 +1,267 @@
+from typing import Union
+
+import numpy as np
+
+from ._text_constants import TextMode, Anchor
+from .text_utils import (
+    format_text_direct,
+    format_text_properties,
+    get_text_anchors,
+)
+from ...utils.colormaps.standardize_color import transform_color
+from ...utils.event import EmitterGroup, Event
+
+
+class TextManager:
+    """ Manages properties related to text displayed in conjunction with the layer
+
+    Parameters
+    ----------
+    text : array or str
+        the strings to be displayed
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
+    anchor : str
+        The location of the text origin relative to the bounding box.
+        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'
+    translation : np.ndarray
+        Offset from the anchor point.
+    color : array or str
+        Font color for the text
+    size : float
+        Font size of the text. Default value is 12.
+    visible : bool
+        Set to true of the text should be displayed.
+
+    Attributes
+    ----------
+    text : array
+        the strings to be displayed
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
+    anchor : str
+        The location of the text origin relative to the bounding box.
+        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
+    translation : np.ndarray
+        Offset from the anchor point.
+    color : array
+        Font color for the text
+    size : float
+        Font size of the text. Default value is 12.
+    visible : bool
+        Set to true of the text should be displayed.
+    """
+
+    def __init__(
+        self,
+        text,
+        n_text,
+        properties={},
+        rotation=0,
+        translation=0,
+        anchor='center',
+        color='black',
+        size=12,
+        visible=True,
+    ):
+
+        self.events = EmitterGroup(
+            source=self,
+            auto_connect=True,
+            text=Event,
+            rotation=Event,
+            translation=Event,
+            anchor=Event,
+            color=Event,
+            size=Event,
+            visible=Event,
+        )
+
+        self.events.block_all()
+        self.rotation = rotation
+        self.anchor = anchor
+        self.translation = translation
+        self.color = color
+        self.size = size
+        self.visible = visible
+
+        self._set_text(text, n_text, properties)
+        self.events.unblock_all()
+
+    @property
+    def text(self):
+        """np.ndarray: the text values to be displayed"""
+        return self._text
+
+    # @text.setter
+    # def text(self, text):
+    #
+    #     self._text = text
```

Comment:
Why is this commented?

Reply:
Ah I was debating if we want to have a `TextManager.text` setter method. It feels a bit weird because we don't just directly pass text values. If we had one, I think we'd have to accept a few different types:

```python
# keep the same text pattern, just update values based on new properties
points.layer.text.text = new_properties

# same properties, new text pattern
new_text_pattern = 'new message, same {property_name}'
point.layer.text.text = new_text_pattern

# new properties, new text pattern
points.layer.text.text = (new_properties, new_text_pattern)
```

Currently, the text setting is taken care of on the layer model using the existing method for setting text values ([`TextManager._set_text()`](https://github.com/napari/napari/blob/501fa2f1a503cf8ee105eecd86c54cc8f909b919/napari/layers/utils/text.py#L101)).

All of that is to say, I would like to delete the `TextManager.text` setter. 😛